### PR TITLE
chore: release v0.5.0

### PR DESCRIPTION
**Pull request by wesk**
Release `v0.5.0`

Important: this PR uses the new release workflow implemented in #568. The new release workflow has fewer manual steps, but still requires human intervention. Here is the current release workflow, which will be improved in future PRs:

1. Add a label to each PR, corresponding with if it is a feature, a bugfix, maintenance, or documentation. 
2. Run `Release Drafter` workflow from GitHub Actions tab: https://github.com/awslabs/syne-tune/actions/workflows/release-drafter.yml
3. Inspect the draft release that the workflow generates here: https://github.com/awslabs/syne-tune/releases
  1. Iterate as needed, updating PR titles and labels as needed, and re-running the workflow, until the release notes look good
4. In a PR (such as the present one): 
    1. Update the `syne_tune/version` file with the latest version number, determined by the draft release: https://github.com/awslabs/syne-tune/releases. If any PRs are tagged with `feature`, then it will be a minor version release, else it will be a patch release.
    2. Update the `CHANGELOG.md` file with the latest changes, copy-pasting from the drafted release.
    3. Contributor names are only included in the github release, so remove them from the CHANGELOG.md file itself
5. In the GitHub release, fill-in the summary of the release at the top of the file, highlighting new contributors and major changes.
6. Merge the PR and create the release in GitHub

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,35 @@ All notable changes to this project will be documented in this file.
  
 The format is based on [Keep a Changelog](http://keepachangelog.com/).
 
+## [0.5.0] - 2023-04-13
```
Comment (wesk) [inline CHANGELOG.md:7]: Because this release contains new features, the version number resolver suggests version `0.5.0`, not `0.4.2`

**Comment by wesk** [comment]
This PR proposes grouping entries in the CHANGELOG.md file and in the GitHub release itself according to the following four headings: 
* New Features
* Documentation Updates
* Bugfixes
* Maintenance

Whereas for previous releases, we have been following this convention: 
- Added
- Changed
- Fixed

"Added" maps to "New Features"; "Fixed" maps to "Bugfixes". 

"Changed" can map to any of the proposed new categories, which attempt to be more specific and actionable for the reader. 

Feedback on this proposed update?

**Comment by mseeger** [comment]
We should before that merge in open PRs, in particular the one I did. This may take a bit of time still.

Not sure about 0.5.0, but why not?

**Comment by mseeger** [comment]
@wesk One more thing to do, is to remove sortedcontainers from requirements.txt files used in benchmarking. This is because for remote launching, we install syne tune from pip, and the sortedcontainers dependence was introduced since the last recent release.

**Comment by mseeger** [comment]
We also need to update "what's new?" in `index.rst` of the docs.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,43 @@ All notable changes to this project will be documented in this file.
  
 The format is based on [Keep a Changelog](http://keepachangelog.com/).
 
+<a name="v0.5.0"></a>
+## [v0.5.0] - 2023-04-20
+
+### New Features
+- simple linear scalarization scheduler ([#619](https://github.com/awslabs/syne-tune/issues/619))
```
Comment (mseeger) [inline CHANGELOG.md:11]: This is still missing #628

File: docs/source/index.rst [hunk 3]
```diff
@@ -31,6 +31,15 @@ This package provides state-of-the-art algorithms for hyperparameter optimizatio
 What's New?
 -----------
 
+* Speculative early checkpoint removal for async multi-fidelity optimization.
```
Comment (mseeger) [inline docs/source/index.rst:34]: I'll edit this.

**Review by mseeger — APPROVED** [review]

File: docs/source/index.rst (hunk 3 above)
Comment (wesk) [reply docs/source/index.rst]: Thank you!

File: CHANGELOG.md (hunk 2 above)
Comment (wesk) [reply CHANGELOG.md]: Thanks for updating it

**Pull request merged**